sympy/integrals/manualintegrate.py

```diff
@@ -486,6 +487,33 @@ def _add_degenerate_step(generic_cond, generic_step, degenerate_step):
     return PiecewiseRule(subfunctions, generic_step.context, generic_step.symbol)
 
 
+def nested_pow_rule(integral: IntegralInfo):
+    # nested (c*(a+b*x)**d)**e
+    integrand, x = integral
+    a = Wild('a', exclude=[x])
+    b = Wild('b', exclude=[x, 0])
+    pattern = a+b*x
+    base = x
+    generic_cond = S.true
+    for add_ in integrand.find(Add):
+        match = add_.match(pattern)
+        if match:
+            a, b = match[a], match[b]
+            base = x + a/b
+            generic_cond = Ne(b, 0)
+            break
+    exp_ = ((base*integrand).diff(x)/integrand).cancel() - 1
+    if exp_.has(x):
+        return
```

Comment:
What happens here if there's more than one `Add`?

Is it really necessary to use `diff` and `cancel` (potentially expensive functions) just to match the pattern `(c*(a+b*x)**d)**e`?

Reply:
> What happens here if there's more than one `Add`?

It doesn't matter. This rule allows at most one `a+b*x` pattern, and the first recognized such pattern is used. For other `Add`s, they are either
* A constant addition, e.g. `integrand == sqrt((1+E)*sqrt(x+1))`. `1+E` won't pass `if match`, so it won't affect anything.
* Another `a+b*x` pattern, e.g. `integrand == sqrt(x+1)*sqrt(x+2)`, or a non-linear addition, e.g. `integrand == sqrt(x+1)+sqrt(x)`. This rule is not applicable in both cases. It won't pass the `exp_.has(x)` check. (I think I should also use `has_free` here)

Reply:
I don't think `diff` is expensive. `cancel` may be, though it's already used several times in `manualintegrate.py`. Without `cancel`, the numeric-exp tests I added can still pass, but the symbolic-exp tests can't.
This pattern can have arbitrary number of nested powers, e.g. `sqrt((x**(S(5)/3))**pi)`, not only 2 as `(c*(a+b*x)**d)**e`. I failed to come up with a clean-and-fast pattern matching rule for that.

Reply:
Both `diff` and `cancel` can be expensive. If they are being used here just for structural pattern matching then I think something simpler should be used. 

Should it not be possible to handle nested powers recursively somehow?

Reply:
Indeed, recursion is simpler than I previously thought. More code than cancel/diff though.